content/token22-in-the-client.md

```diff
@@ -0,0 +1,470 @@
+	await mintTo(
+		connection,
+		payer,
+		mint,
+		tokenAccount.address,
+		payer,
+		mintAmount,
+		[payer],
+		{commitment: 'finalized'},
+		tokenProgramId
+	)
+
+	return mint
+}
+
+export default createAndMintToken
+```
+
+
+
+### 5. Creating and minting tokens
+
+Let's now take our new function and invoke it twice within our main script in `index.ts`. We'll want a `Token Program` and `Token Extension Program` token to test against. So we'll use our two different program IDs:
+
+```tsx
+import { initializeKeypair } from '@solana-developers/helpers';
+import { Cluster, Connection } from '@solana/web3.js';
+import createAndMintToken from './create-and-mint-token';
+import printTableData from './print-helpers';
+import { TOKEN_2022_PROGRAM_ID, TOKEN_PROGRAM_ID } from '@solana/spl-token';
+import dotenv from 'dotenv';
+dotenv.config();
+
+(async () => {
+	/**
+	 * Create a connection and initialize a keypair if one doesn't already exists.
+	 * If a keypair exists, airdrop a sol if needed.
+	 */
+	const connection = new Connection('http://127.0.0.1:8899', 'confirmed')
+	const payer = await initializeKeypair(connection)
+
+	console.log(`Payer: ${payer.publicKey.toBase58()}`)
+
+	const tokenProgramMint = await createAndMintToken(
+		connection,
+		TOKEN_PROGRAM_ID,
+		payer,
+		0,
+		1000
+	)
+	const tokenExtensionProgramMint = await createAndMintToken(
+		connection,
+		TOKEN_2022_PROGRAM_ID,
+		payer,
+		0,
+		1000
+	)
+})();
+```
+
+At this point you can run `npm run start` and see that both mints get created and their info logged to the console.
+
+### 6. Fetch Token Program and Token Extension Program tokens
+
+We can now fetch tokens using the wallet's public key and the program ID.
+
+Let's create a new file `fetch-token-info.ts`.
+
+Within that new file, let's create the `fetchTokenInfo` function. This function will fetch the token account provided and return a new interface we'll create named `TokenInfoForDisplay`. This will allow us to format the returning data nicely in our console. Again, this function will be agnostic about which token program the account it from.
+
+```ts
+import { AccountLayout, getMint } from "@solana/spl-token"
+import { Connection, LAMPORTS_PER_SOL, PublicKey } from "@solana/web3.js"
+
+export type TokenTypeForDisplay = 'Token Program' | 'Token Extension Program';
+
+export interface TokenInfoForDisplay {
+  mint: PublicKey
+  amount: number
+  decimals: number
+  displayAmount: number
+  type: TokenTypeForDisplay
+}
+
+```
+
+To actually fetch all of this information we will be calling `getTokenAccountsByOwner` and mapping the results into our new `TokenInfoForDisplay` interface.
+
+To accomplish this the `fetchTokenInfo` function will need the following parameters:
+
+- `connection` - The connection object to use
+- `owner` - The wallet which owns the associated token accounts
+- `programId` - The token program to point to
+- `type` - Either `Token` or `Token22`; used for console logging purpose
```

Comment:
Looking at the code I think this is "Token Program' | 'Token Extension Program'" (which is better!)

Reply:
You are correct!